### PR TITLE
[3/6] Venft approve and mint

### PR DESCRIPTION
**Pull request by jmilldotdev**
## What does this PR do and why?

Adds the complete flow for approving the veNft contract to spend the project ERC-20 token, and for locking and minting the veNft using the data in the form.

## Screenshots or screen recordings

image

Unapproved wallet

image

Minting a token for an approved wallet.

(Note: fetching the data for the image is absent from the confirmation modal right now, will be added in the next PR.)

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/veNft/formControls/StakingFormActionButton.tsx [hunk 1]
```diff
@@ -31,7 +31,12 @@ const StakingFormActionButton = ({
     }
     if (!hasAdequateApproval) {
       return (
-        <Button block style={{ whiteSpace: 'pre' }} onClick={approve}>
+        <Button
+          block
+          style={{ whiteSpace: 'pre' }}
```
Comment (tomquirk) [inline src/components/veNft/formControls/StakingFormActionButton.tsx:36]: **quesiton** do we need this?

File: src/components/veNft/formControls/StakingFormActionButton.tsx [hunk 2]
```diff
@@ -31,7 +31,12 @@ const StakingFormActionButton = ({
     }
     if (!hasAdequateApproval) {
       return (
-        <Button block style={{ whiteSpace: 'pre' }} onClick={approve}>
+        <Button
```
Comment (tomquirk) [inline src/components/veNft/formControls/StakingFormActionButton.tsx:34]: **suggestion** think we can add `type="primary"` here

File: src/components/veNft/formControls/StakingFormActionButton.tsx [hunk 3]
```diff
@@ -31,7 +31,12 @@ const StakingFormActionButton = ({
...
+          block
+          style={{ whiteSpace: 'pre' }}
+          onClick={onApproveButtonClick}
+          loading={tokenApprovalLoading}
+        >
           Approve Token for Transaction
```
Comment (tomquirk) [inline src/components/veNft/formControls/StakingFormActionButton.tsx:40]: **suggestion** for UI copy, we tend to lowercase everything.

```suggestion
          Approve token for transaction
```

**Review by tomquirk — COMMENTED** [review]
@jmilldotdev 

### UX

- WDYT about hiding the form before the user has "Approved token for txn"? The user might fill out the form
- Does the whole page have to reload after the user "Approved token for txn"? If the tx succeeds, we can optimistically _assume_ that the "stake" tx will work. In other words, we can immediately update the UI when the "approve" tx succeeds.
-  There's a mixture of "stake" and "lock" terminology. Should we decide on one term?

File: src/components/veNft/veNftConfirmStakeModal.tsx [hunk 4]
```diff
@@ -1,28 +1,144 @@
...
+        </Trans>
+      </Callout>
+      <Row>
+        <Col span={14}>
+          <Descriptions
+            title={
+              <h3>
+                <Trans>$ve{tokenSymbolDisplayText} NFT summary:</Trans>
+              </h3>
+            }
```
Comment (tomquirk) [inline src/components/veNft/veNftConfirmStakeModal.tsx:117]: **suggestion** feel like we could remove this. wdyt?

File: src/components/veNft/formControls/StakingFormActionButton.tsx (hunk 1 above)
Comment (jmilldotdev) [reply src/components/veNft/formControls/StakingFormActionButton.tsx:36]: this was addressed in a downstream branch

File: src/components/veNft/formControls/StakingFormActionButton.tsx (hunk 2 above)
Comment (jmilldotdev) [reply src/components/veNft/formControls/StakingFormActionButton.tsx:34]: addressed in downstream branch

File: src/components/veNft/veNftConfirmStakeModal.tsx (hunk 4 above)
Comment (jmilldotdev) [reply src/components/veNft/veNftConfirmStakeModal.tsx:117]: SGTM

File: src/components/veNft/formControls/StakingFormActionButton.tsx (hunk 2 above)
Comment (jmilldotdev) [reply src/components/veNft/formControls/StakingFormActionButton.tsx:34]: done

**Comment by jmilldotdev** [comment]
All changes addressed

**Pull request merged**